pkg/dnscache/nodednscache.go

```diff
@@ -0,0 +1,139 @@
+package dnscache
+
+import (
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/hashicorp/golang-lru/v2/expirable"
+
+	"github.com/aquasecurity/tracee/types/detect"
+	"github.com/aquasecurity/tracee/types/trace"
+)
+
+type NodeCache struct {
+	queryRoots   *expirable.LRU[string, *dnsQueryNode]
+	queryIndices map[string]*dnsQueryNode
+
+	lock *sync.RWMutex
+}
+
+func NewNodeDNSCache(config Config) *NodeCache {
+	ttl := time.Second * time.Duration(config.TTL)
+	nc := &NodeCache{}
+	nc.queryRoots = expirable.NewLRU[string, *dnsQueryNode](config.CacheSize, nc.evictQueryRoot, ttl)
+	nc.queryIndices = make(map[string]*dnsQueryNode)
+	nc.lock = new(sync.RWMutex)
+	return nc
+}
+
+func (nc *NodeCache) Add(dns *trace.ProtoDNS) error {
```

Comment:
@rafaeldtinoco this implementation will use `net_packet_dns` instead of the legacy event like you wanted for the original one.